src/main/java/com/laserfiche/repository/api/RepositoryApiClientImpl.java

```diff
@@ -57,6 +54,15 @@ private <C> C createClient(Class<C> clientInterface) {
                 .create(clientInterface);
     }
 
+    private void setDefaultHeaders(String servicePrincipalKey, AccessKey accessKey) {
```

Comment:
This method is called `setDefaultHeader` but its content doesn't really set the header. Maybe a better name?

Reply:
I think overall the `setLoadTestHeader` and `setDefaultHeaders` feel a bit out of place for me. Can you give me some time to see if there's a better alternative?

Reply:
how about `setDefaultHeaderInterceptor` or `setLoadTestHeaderInterceptor`?